client.go

```diff
@@ -436,16 +438,36 @@ func (c *client) sendQuery(msg *dns.Msg) error {
 		return err
 	}
 	if c.ipv4conn != nil {
+		// See https://pkg.go.dev/golang.org/x/net/ipv4#pkg-note-BUG
+		// As of Golang 1.18.4
+		// On Windows, the ControlMessage for ReadFrom and WriteTo methods of PacketConn is not implemented.
 		var wcm ipv4.ControlMessage
 		for ifi := range c.ifaces {
-			wcm.IfIndex = c.ifaces[ifi].Index
+			switch runtime.GOOS {
+			case "darwin", "ios", "linux":
```

Comment:
Why is this special-casing darwin, ios and linux, and not windows?

Reply:
As I was looking at all the examples of using SetMulticastInterface I saw this example on Google's own Golang docs
https://pkg.go.dev/golang.org/x/net/ipv4#example-RawConn-AdvertisingOSPFHello

I believe it comes down to ControlMessage only being implemented on linux/iOS/darwin mac based platforms and none others. It didn't make sense to specifically call out windows. See https://pkg.go.dev/golang.org/x/net/ipv4#PacketConn.WriteTo